packages/whenable/src/types.js

```diff
@@ -0,0 +1,50 @@
+// @ts-check
+export {};
+
+/**
+ * @template T
+ * @typedef {PromiseLike<T | Whenable<T>>} PromiseWhenable
+ */
+
+/**
+ * @template [T=any]
+ * @typedef {{ whenable0: { shorten(): Promise<T | Whenable<T>>} }} WhenablePayload
```

Comment:
As a property name here, we'd be committing to the name `whenable0` for a long time. What does the zero mean?

Reply:
Since it contains a locally-synchronously-callable function, a `Whenable`, though encoded in a CopyTagged, could not actually be passed. While I understand you may not need to pass this between vats, how are you gonna restore it from durable storage?

https://github.com/endojs/endo/issues/1583 explains the conflict between separately validatable data structures and allegedly well behaved methods. By placing this function within a WhenablePayload, does it have this problem as well? What happens when it meets its eval twin?

Reply:
Deepening my confusion: To even have this pass `passStyleOf`, either the `shorten` method would need to be a far function or its containing object would need to be a far object.

Reply:
(Confusion resolved below, so nevermind. Leaving confused trail just fyi. Close when not interesting. FWIW, what confused me was expanding a far object's type inline in a containing type with no indication it is a remotable.)

Reply:
> no indication it is a remotable

That's a good point.  I should improve this to be a `FarRef`.

Reply:
> we'd be committing to the name `whenable0` for a long time.  What does the zero mean?

I was thinking that it would mean the 0th version of the whenable wire protocol, with more to come (such as one that offers eventual pipelining).  I could have it be something less opaque (like `shortener`), if you deem that appropriate.  In either event, I will document the meaning in a new type with some commentary explaining that its methods must never be modified.

Reply:
For the interim, I changed `whenable0` to `whenableV0`.

Reply:
Good. That "V" makes it much easier to guess what the name means ;)

Reply:
I notice that passing a Whenable over the wire violates the type you originally wrote down, because it arrives with the `whenableV0` value being a remote presence for the far object, rather than the far object itself. I'm pleased to see that
- It does not violate the new type
- This case is compatible with your two existing uses of `.shorten()`, as they are both eventual sends to this far object.

Resolving.